DSharpPlus/Entities/Channel/DiscordChannel.cs

```diff
@@ -405,14 +405,17 @@ public async Task<DiscordChannel> CloneAsync(string reason = null)
         /// Returns a specific message
         /// </summary>
         /// <param name="id">The id of the message</param>
+        /// <param name="useCacheIfPossible">True to return a cached object if possible, false to always retrieve a fresh object from Discord.</param>
         /// <returns></returns>
         /// <exception cref="UnauthorizedException">Thrown when the client does not have the <see cref="Permissions.ReadMessageHistory"/> permission.</exception>
         /// <exception cref="NotFoundException">Thrown when the channel does not exist.</exception>
         /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
         /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
-        public async Task<DiscordMessage> GetMessageAsync(ulong id)
+        /// <remarks>Cached message objects will not be returned if <see cref="DiscordConfiguration.MessageCacheSize"/> is set to zero, if the client does not have the <see cref="DiscordIntents.GuildMessages"/> or <see cref="DiscordIntents.DirectMessages"/> intents, or if the discord client is a <see cref="DiscordShardedClient"/>.</remarks>
+        public async Task<DiscordMessage> GetMessageAsync(ulong id, bool useCacheIfPossible = true)
```

Comment:
ditto

Reply:
Can't really update this cache as it's a RingBuffer, but I'll invert the parameter everywhere and make it update the cache if true.